dash/App/Chains/Connection/index.js

```diff
@@ -29,23 +41,33 @@ class ChainModule extends React.Component {
   //   document.removeEventListener('click', this.clickHandler.bind(this))
   // }
 
-  renderConnection (origin, id) {
+  renderConnection (id, { primary, secondary }, blockHeight) {
+    let currentConnectionName = ''
+    let currentConnectionStatus = ''
+    if (primary.on && primary.status !== 'disconnected') {
```

Comment:
mostly looks good, still not sure about this logic though. it looks like in chains.js it does not check for "on" and only checks for "connected". for example if the primary connection is not connected it will fall back to the seconday. however your logic if the primary connection is "on" but not connected (say, pending) we will display that even if the secondary connection is connected.

Reply:
also why do we show the status as the name when the connection is not connected? can't we still show the chain name?

Reply:
this is the logic for the current version where we show the currently selected chain. it looks like if either the primary or secondary connection is "connected" it will show as green/connected

https://github.com/floating/frame/blob/0.5/app/App/Panel/index.js#L88

Reply:
Updated for your second point.

Reply:
I see what you get about the original logic, but I think ignoring the pending state completely is not the way to go.

Reply:
Ok, updated again.  If this isn't it then I think we need to look at each individual state combination in a table and thrash out a solution.